### PR TITLE
MfsApi implementation of IMfsApi for making calls to /files endpoints

### PR DESCRIPTION
**Pull request by Derrick-**
Implements IMfsApi for interacting with MFS via the Core API. This includes all endpoints under /api/v0/files/.

Tested against ipfs version 0.9.0

Requires: https://github.com/ipfs-shipyard/net-ipfs-core/pull/13

Related:
https://github.com/ipfs-shipyard/net-ipfs-http-client/issues/14
https://github.com/ipfs-shipyard/net-ipfs-core/issues/11

### CODE REVIEW

**Comment by Derrick-** [comment]
There is a bug in this PR.
ListAsync FileSystemNode results do not include the active IpfsClient field.
Will fix.

**Comment by Derrick-** [comment]
@Arlodotexe if there's anything I can do to help the review process on this, or if you are uncomfortable with the way it's implemented please let me know.
I'm interested in knocking out some other issues, but waiting on the outcome this one before proceeding

**Comment by Arlodotexe** [comment]
> @Arlodotexe if there's anything I can do to help the review process on this, or if you are uncomfortable with the way it's implemented please let me know. I'm interested in knocking out some other issues, but waiting on the outcome this one before proceeding

Hey @Derrick-, sorry for the delay. I've been caught up in many other projects, but I'm happy to help you keep this moving. let's start by getting your prerequisite PR done and published to nuget -> https://github.com/ipfs-shipyard/net-ipfs-core/pull/13

**Comment by Arlodotexe** [comment]
@Derrick- `IMfsApi` was published in [IpfsShipyard.Ipfs.Core](https://www.nuget.org/packages/IpfsShipyard.Ipfs.Core) 0.0.2. You should be able to complete this PR now.

**Comment by Derrick-** [comment]
Thanks. wasn't trying to rush you

> @Derrick- `IMfsApi` was published in [IpfsShipyard.Ipfs.Core](https://www.nuget.org/packages/IpfsShipyard.Ipfs.Core) 0.0.2. You should be able to complete this PR now.

Thanks. I don't have permission to merge in this repo though, you will need to approve and merge please.

**Comment by Arlodotexe** [comment]
> > @Derrick- `IMfsApi` was published in [IpfsShipyard.Ipfs.Core](https://www.nuget.org/packages/IpfsShipyard.Ipfs.Core) 0.0.2. You should be able to complete this PR now.
> 
> Thanks. I don't have permission to merge in this repo though, you will need to approve and merge please.

The build errors need to be fixed first. Should just be updating the `net-ipfs-core` package at this point,

File: src/IpfsHttpClient.csproj [hunk 1]
```diff
@@ -46,7 +46,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="IpfsShipyard.Ipfs.Core" Version="0.0.1" />
+    <PackageReference Include="IpfsShipyard.Ipfs.Core" Version="0.0.4" />
```
Comment (Arlodotexe) [inline src/IpfsHttpClient.csproj:49]:
```suggestion
    <PackageReference Include="IpfsShipyard.Ipfs.Core" Version="0.0.5" />
```

**Review by Arlodotexe — APPROVED** [review]

**Pull request merged**